python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -0,0 +1,258 @@
+            if walkDist <= radiusinmeters: #within 0.5 miles walking
+                distCsvWriter.writerow((Nlihc_id, 'bus', stop['StopID'], "{0:.2f}".format(self.getMiles())))
+
+class railStation():
+
+    def __init__(self,wmata_api_key,mapbox_api_key):
+        self.wmata_api_key = wmata_api_key
+        self.mapbox_api_key = {'access_token':mapbox_api_key}
+
+    def getWmataHeaders(self):
+        return { 'api_key': self.wmata_api_key}
+
+    def writeRailInfo(self, infoCsvWriter):
+        """Writes all rail station data to a given CSV writer. Returns the railStations json for future processing
+
+           Parameters:
+           infoCsvWriter - csv writer
+           wmata_api_key - api key for wmata REST services
+           """
+        print("Writing RAIL INFO")
+
+        wmata_headers = self.getWmataHeaders()
+
+        railResponse = requests.get("https://api.wmata.com/Rail.svc/json/jStations", headers=wmata_headers)
+        railStations = railResponse.json()['Stations']
+
+        for station in railStations:
+            #delimit list of lines with colon
+            lines = station["LineCode1"] #there is always at least one station
+            for line_code in ["LineCode2", "LineCode3", "LineCode4"]:
+                if station[line_code] != None:
+                    lines += ":" + station[line_code]
+            infoCsvWriter.writerow((station['Code'], 'rail',station['Name'],str(station['Lat']), str(station['Lon']),lines))
+
+        return railStations
+
+    def writeBusInfo(self, infoCsvWriter):
+        """Writes all bus station data to a given CSV writer.
+
+            Parameters:
+            infoCsvWriter - csv writer
+            wmata_api_key - api key for wmata REST services
+            """
+
+        print("Writing BUS INFO")
+
+        wmata_headers = self.getWmataHeaders()
+
+        response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', headers=wmata_headers)
+        data = response.json()
+
+        for stop in data['Stops']:
+
+            lines = ""
+            for route in stop['Routes']:
+                lines = '{}:{}'.format(lines, route)
+            lines = lines[1:] #take off the first :
+
+            infoCsvWriter.writerow((stop['StopID'], 'bus', stop['Name'], stop['Lat'],stop['Lon'], lines)) #, stop_id_or_station_code))
+    
+
+def main(secretsFileName, distOutputFileName,infoOutputFileName):
+    """Writes two csvs: 1 for general bus/rail info, 1 with distances to wmata for projects
+
+   Parameters:
+   secretsFileName - json file name that contains various api keys
+   csvInputFileName - csv file with project information
+   distOutputFileName - csv file to output to for calculated metro distances for each project
+   infoOutputFileName - cvs file for general bus & rail info for each wmata station
+   """
+
+    try:
+        conn = psycopg2.connect("dbname='housinginsights_docker' user='codefordc' host='192.168.99.100' password='codefordc'")
```

Comment:
Oh, also - should use the sqlAlchemy connection instead of psycopg2, just since that's what we're doing elsewhere. That's what that funciton returns.